### ISSUE TITLE
Pass color-contrast if aria-model="true" is on the page, and the text is aria-hidden

### ISSUE THREAD

**Issue by smhigley** [dequelabs/axe-core#1813]
Related to #170

Elements in the background of modals are currently flagged as contrast issues, even when within an `aria-hidden` parent. Example here: https://jsfiddle.net/204dkrsn/2/show

I know the applicability of contrast to `aria-hidden` elements is up for interpretation, since they're also often visible. On the other hand, the current setup has been generating some false positives when modals are open.

Another workaround might be ignoring visual rules on anything outside an `aria-modal="true"` element if one exists on the page, but I'm not sure how feasible that is.

**Comment by straker** [comment]
Thanks for the issue. This is a similar problem we encountered in #1482. Could probably solve them in the same way.

**Comment by straker** [comment]
I ran the example and the results are returning as incomplete with `"Element's background color could not be determined because it is overlapped by another element"` since the modal is on top. Is this what you are seeing?

**Comment by smhigley** [comment]
I get this error in Firefox:

```
Fix the following:

Element has insufficient color contrast of 2.01 (foreground color: #000000, background color: #404040, font size: 12.0pt, font weight: normal). Expected contrast ratio of 4.5:1
```

After your comment I checked Chrome, where I see the same warning as you. Chrome is using axe-core 4.0.0, and Firefox has 3.2.2, so perhaps it's a version issue?

**Comment by straker** [comment]
It seems to be. axe 3.2.2 returns with an error while axe 3.3.2 returns as incomplete. I would hope both Firefox and Chrome are using the latest version, so I'll have to see what's going on.

**Comment by WilcoFiers** [comment]
I think we can make the change, so if there's an aria-modal="true", the text node is in an aria-hidden="true" region, it is overlapped with a semi-transparent node, we can exclude it from the needs review. That seems reasonable to me. I don't think this is a particularly high priority thing to sort out, but it would be a nice addition, to cut down on the needs review stuff in color-contrast.

**Comment by straker** [comment]
@smhigley The Firefox extension has been updated to 3.3.2

**Comment by smhigley** [comment]
@straker sorry I missed this, thanks so much! I'm closing the issue, since I think the behavior is correct in both FF and Chrome now.

Thanks again!
